app/src/main/java/com/techcourse/service/RegisterService.java

```diff
@@ -0,0 +1,36 @@
+package com.techcourse.service;
+
+import com.techcourse.controller.LoginController;
+import com.techcourse.controller.request.RegisterRequest;
+import com.techcourse.dao.UserDao;
+import com.techcourse.domain.User;
+import com.techcourse.exception.DuplicateAccountException;
+import nextstep.web.annotation.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+public class RegisterService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LoginController.class);
+
+    private final UserDao userDao;
+
+    public RegisterService(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+    public void registerUser(RegisterRequest request) {
+        User user = request.toEntity();
+        validateDuplicate(user);
+
+        userDao.insert(user);
+    }
+
+    private void validateDuplicate(User user) {
+        if (userDao.findByAccount(user.getAccount()).isPresent()) {
+            LOG.debug("Duplicate account already exist => {}", user.getAccount());
+            throw new DuplicateAccountException();
```

Comment:
```suggestion
        userDao.findByAccount(user.getAccount())
                .ifPresent(findUser -> new DuplicateAccountException());
```

이 내용이 지금 상황에도 해당이 될지는 모르겠지만... [[Effective Java] item 55. 옵셔널 반환은 신중히 하라](https://hirlawldo.tistory.com/103)에서는 `isPresent` 사용을 피할 수 있는 경우에는 피하라고 하고있어요 😃
그 이유는 [해당 포스터의 Optional의 잘못된 사용 예제](https://www.daleseo.com/java8-optional-after/)를 살펴보면 아실 수 있을 것 같아요 😆   (사실 이 내용을 제가 잘 못 이해하고있는지 모르겠지만 `isPresent`를 사용하기 전에는 다른 방식으로 대체할 수 없을까 항상 고민하는 편이에요!! )